nexus/db-model/src/instance.rs

```diff
@@ -179,6 +179,18 @@ pub struct InstanceRuntimeState {
     /// This field is guarded by the instance's `gen`.
     #[diesel(column_name = migration_id)]
     pub migration_id: Option<Uuid>,
+
+    /// The UUID of the saga currently holding the update lock on this instance.
+    /// If this is [`None`] the instance is not locked. Otherwise, if this is
+    /// [`Some`], this contains the UUID of the saga holding the lock.
+    ///
+    /// This field is guarded by the instance's `updater_gen`
+    #[diesel(column_name = updater_id)]
+    pub updater_id: Option<Uuid>,
```

Comment:
I want to be super specific about this: What is the update lock protecting against? Right now, this PR looks like it's using it to ensure there is only one active saga for destroying an instance. However, is it necessary to "try lock" access to this lock for other operations on the instance that might also be happening concurrently?

E.g. What stops someone from concurrently rebooting the instance while we're trying to grab the "update lock"? Could that operation run to completion if it happens before `DELETE_SLED_RESOURCE` in the `instance_update_destroyed` saga?